### PR TITLE
fix!: Unify icon usage

### LINKED ISSUE #1303: [feat] Unify icon usage
**Does your feature request relate to a specific USWDS component?**

Dependent on new react-uswds Icon component. PR: https://github.com/trussworks/react-uswds/pull/1241
More context: https://github.com/trussworks/react-uswds/issues/992

**What USWDS Version is this feature present in?**
USWDS 2.11 +

**Is your feature request related to a problem? Please describe.**
USWDS has moved away from Font Awesome towards Google Material Icons, as well as using SVG icon sprite. Some of the changes are discussed in this [video](https://youtu.be/OhtiTfKAIRw?t=2843). We will follow suit and update this codebase as we move to 2.11. This will require making some documentation edits as well as using our new [Icon ](https://designsystem.digital.gov/components/icon/) component.

**Describe the solution you'd like**
Several steps to take here:
- [x] [Make sure react-uswds is on 2.11+](https://github.com/trussworks/react-uswds/issues/1304)
- [ ] Double check our components are using the correct uswds icons. `GovBanner` uses a lock icon for example.
- [ ] Replace usage of icons in our components with our react-uswds `<Icon>`. 
- [x] In our styles and assets documentation we're recommend use fontawesome per uswds' previous recommendation. Our documentation needs to be updated here to remove reference to fontawesome in favor of Material icons.
- [x] Remove usage of @fortawesome fontawesome in example app entirely
- [ ] USWDS examples set some intrinsic props on `<svg/>` tags to defaults (e.g. focusable and role). We should set the same defaults in our code,  but allow override these html attributes if the corresponding prop is passed to the Icon component

### PR DESCRIPTION
**Pull request by werdnanoslen**
# Summary

Addresses these two ACs in the original issue:
- Double check our components are using the correct uswds icons. GovBanner uses a lock icon for example.
- Replace usage of icons in our components with our react-uswds <Icon>.

## Related Issues or PRs

Resolves #1303

## How To Test

Confirm that all SVGs still in use are for images that aren't icons

### Screenshots (optional)
N/A, looks the same visually, and behaves the same with some attribute changes

### CODE REVIEW

**Comment by werdnanoslen** [comment]
Social links I wasn't sure if it needed to change per se, given the styles referenced there turn it black based on its parent div and USWDS doesn't define their style directly. So since our implementation uses their styles, they are sort of 'supposed' to be blue out of their context, since they're links. I _think_ 🤷 

I'll fix the search though, could've sworn they were the same on my local. 🤔

**Comment by werdnanoslen** [comment]
Ok yeah search definitely looks normal on my local in both Firefox & Chrome, 24px icon with 12px margin
image
image

**Comment by werdnanoslen** [comment]
@brandonlenz are Happo diffs approvable?

File: src/components/Footer/SocialLinks/SocialLinks.test.tsx [hunk 1]
```diff
@@ -40,7 +40,9 @@ describe('SocialLinks component', () => {
 describe('SocialLink component', () => {
   it('renders without errors', () => {
     const { container } = render(<SocialLink name="Instagram" />)
-    expect(container.querySelector('.usa-social-link')).toBeInTheDocument()
-    expect(screen.getByRole('img')).toHaveAttribute('alt', 'Instagram')
```
Comment (gidjin) [inline src/components/Footer/SocialLinks/SocialLinks.test.tsx:44]: Question: Should these have alt tags too?

**Review by gidjin — APPROVED** [review]
LGTM, left a question but it's not blocking.

File: src/components/Footer/SocialLinks/SocialLinks.test.tsx (hunk 1 above)
Comment (werdnanoslen) [reply src/components/Footer/SocialLinks/SocialLinks.test.tsx:44]: no, the `alt` prop only applies to ``. Now that it's an `<svg>`, I have a line at the bottom of that block to check the `name` prop instead, which serves the same purpose.

**Pull request merged**